### PR TITLE
feat: create setting in merge widget to set status as pending in failing situations

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

**Comment by ocombe** [comment]
Thanks but I think that it should be more subtle than that. For example when another status is failed, it might as well be failed for this one as well.
I've discussed it with Jeremy and he would like to only check some labels when the merge label has been applied. I'll update your PR

**Comment by ocombe** [comment]
What do you think of my changes?

File: functions/src/plugins/merge.ts [hunk 1]
```diff
@@ -82,7 +82,10 @@ export class MergeTask extends Task {
       updateG3Status = true;
     }
 
-    if(matchAny([newLabel], config.checks.requiredLabels) || matchAny([newLabel], config.checks.forbiddenLabels)) {
+    if(matchAny([newLabel], config.checks.requiredLabels)
+      || matchAny([newLabel], config.checks.forbiddenLabels)
+      || (getLabelsNames(labels).indexOf(config.mergeLabel) !== -1 && matchAny([newLabel], config.checks.requiredLabelsWhenMergeReady))
```
Comment (jelbourn) [inline functions/src/plugins/merge.ts]: Use `.includes()` instead of `indexOf()`?

(here and below)

File: functions/src/plugins/merge.ts [hunk 2]
```diff
@@ -108,7 +111,11 @@ export class MergeTask extends Task {
       throw err;
     });
 
-    if(matchAny([removedLabel], config.checks.requiredLabels) || matchAny([removedLabel], config.checks.forbiddenLabels)) {
+    if(
+      matchAny([removedLabel], config.checks.requiredLabels)
```
Comment (jelbourn) [inline functions/src/plugins/merge.ts]: Break this check into a function since it's used more than once?

File: functions/src/plugins/merge.ts [hunk 3]
```diff
@@ -174,7 +181,21 @@ export class MergeTask extends Task {
...
+    if(labelsNames.indexOf(config.mergeLabel) !== -1 && config.checks.requiredLabelsWhenMergeReady) {
+      const missingLabels = [];
+      config.checks.requiredLabelsWhenMergeReady.forEach(reqLabel => {
+        if(!labelsNames.some(label => !!label.match(new RegExp(reqLabel)))) {
+          missingLabels.push(reqLabel);
+        }
+      });
+
+      if(missingLabels.length > 0) {
+        checksStatus.pending.push(`missing required label${missingLabels.length > 1 ? 's' : ''}: "${missingLabels.join('", "')}"`);
```
Comment (jelbourn) [inline functions/src/plugins/merge.ts]: Could eliminate the ternary by phrasing this like
```ts
`Missing required labels: ${missingLabels.join(', ')}`
```

**Pull request merged**